mergify_engine/rules/__init__.py

```diff
@@ -247,28 +258,31 @@ async def create(
             # * rules that won't work due to a configuration issue detected at runtime (eg team doesn't exists) -> faulty_rules
             # * rules where only BASE_ATTRIBUTES don't match (mainly to hide rule written for other branches) -> ignored_rules
             categorized = False
-
+            evaluated_rule = typing.cast(T_EvaluatedRule, rule)
             if hide_rule and not rule.conditions.match:
                 # NOTE(sileht): Replace non-base attribute by true, if it
                 # still matches it's a potential rule otherwise hide it.
                 base_conditions = rule.conditions.copy()
                 for condition in base_conditions.walk():
                     attr = condition.get_attribute_name()
-                    if attr not in self.BASE_ATTRIBUTES:
+                    if (attr not in self.BASE_ATTRIBUTES) and (
+                        attr not in self.BASE_CHANGEABLE_ATTRIBUTES
+                    ):
                         condition.update("number>0")
 
                 await base_conditions(pulls)
 
                 if not base_conditions.match:
-                    self.ignored_rules.append(typing.cast(T_EvaluatedRule, rule))
+                    self.base_changeable_attributes_rules.append(evaluated_rule)
+                    self.ignored_rules.append(evaluated_rule)
```

Comment:
The two lists look to be the same, no?

I think you can directly build the `not_applicable_base_changeable_attributes_rules` here?